pkg/sdk/manifest/json_type.go

```diff
@@ -0,0 +1,49 @@
+package manifest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"capact.io/capact/pkg/sdk/apis/0.0.1/types"
+	"github.com/pkg/errors"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+// TypeValidator is a validator for Type manifest.
+type TypeValidator struct{}
+
+// NewTypeValidator creates new TypeValidator.
+func NewTypeValidator() *TypeValidator {
+	return &TypeValidator{}
+}
+
+// Do is a method which triggers the validation.
+func (v *TypeValidator) Do(_ context.Context, _ types.ManifestMetadata, jsonBytes []byte) (ValidationResult, error) {
+	var typeEntity types.Type
+	err := json.Unmarshal(jsonBytes, &typeEntity)
+	if err != nil {
+		return ValidationResult{}, errors.Wrap(err, "while unmarshalling JSON into Type type")
+	}
+
+	jsonSchemaStr := typeEntity.Spec.JSONSchema.Value
+	schemaLoader := gojsonschema.NewReferenceLoader("http://json-schema.org/draft-07/schema")
+	manifestLoader := gojsonschema.NewStringLoader(jsonSchemaStr)
+
+	jsonSchemaValidationResult, err := gojsonschema.Validate(schemaLoader, manifestLoader)
+	if err != nil {
+		return newValidationResult(err), nil
+	}
+
+	result := ValidationResult{}
+	for _, err := range jsonSchemaValidationResult.Errors() {
+		result.Errors = append(result.Errors, fmt.Errorf("%v", err.String()))
```

Comment:
IMO it will be good to add more context here. What we have right now is:
```
- ✗ "../hub-manifests/manifests/core/type/platform/kubernetes.yaml":
    * TypeValidator: invalid character 'd' in string escape code
```
TBH after first look I wasn't able to tell what should be fixed. WDYT about sth like:
```
- ✗ "../hub-manifests/manifests/core/type/platform/kubernetes.yaml":
    * TypeValidator: invalid JSONSchema: invalid character 'd' in string escape code
```
or
```
- ✗ "../hub-manifests/manifests/core/type/platform/kubernetes.yaml":
    * TypeValidator: field 'spec.jsonSchema.value': invalid character 'd' in string escape code
```
or
```
- ✗ "../hub-manifests/manifests/core/type/platform/kubernetes.yaml":
    * TypeValidator: spec.jsonSchema.value: invalid character 'd' in string escape code
```
same as we have for other check:
```
   * OCFSchemaValidator: spec.additionalInput.typeInstances.aws-credentials: verbs is required
```